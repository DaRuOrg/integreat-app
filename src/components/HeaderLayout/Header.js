import React from 'react'
import { NavLink } from 'react-router-dom'
import PropTypes from 'prop-types'
import cx from 'classnames'
import FontAwesome from 'react-fontawesome'
import HeaderDropDown from './HeaderDropDown'

import style from './Header.css'
import helper from 'components/Helper/Helper.css'
import logoWide from './assets/integreat-app-logo.png'
import logoSquare from './assets/integreat-logo-square.png'
import LanguageFlyout from 'components/LanguageFlyout'
import Navigation from 'Navigation'

class NavElement extends React.Component {
  static propTypes = {
    to: PropTypes.string.isRequired,
    className: PropTypes.string,
    disableActiveStyle: PropTypes.bool
  }

  render () {
    return (
      <NavLink exact to={this.props.to} activeClassName={this.props.disableActiveStyle ? '' : style.itemActive}
               className={cx(this.props.className, helper.removeA)}>
        {this.props.children}
      </NavLink>
    )
  }
}

class Header extends React.Component {
  static propTypes = {
    languages: PropTypes.arrayOf(PropTypes.instanceOf(LanguageModel)).isRequired,
    languageCallback: PropTypes.func,
    navigation: PropTypes.instanceOf(Navigation).isRequired,
    currentLanguage: PropTypes.string
  }

  render () {
    return (
      <header className={style.spacer}>
        <div className={style.header}>
          { /* Logo */}
          <NavElement to={this.props.navigation.home} className={style.logo} disableActiveStyle={true}>
            <img className={style.logoWide}
                 src={logoWide}/>
            <img className={style.logoSquare}
                 src={logoSquare}/>
          </NavElement>
          <div className={style.itemsContainer}>
            { /* Home for small devices */}
            <NavElement to={this.props.navigation.home} className={cx(style.item, style.itemHome)}>
              <FontAwesome name='home'/>
            </NavElement>
            { /* Location */}
            <NavElement to={this.props.navigation.search} className={cx(style.item, style.itemSearch)}>
              <FontAwesome name='search'/>
            </NavElement>
            <NavElement to={this.props.navigation.locationSelection} className={cx(style.item, style.itemLocation)}>
              <FontAwesome name='map-marker'/>
            </NavElement>
            { /* Language */}
            <HeaderDropDown className={style.itemLanguage} fontAwesome="globe">
              <LanguageFlyout
                languageCallback={this.props.languageCallback}
                languages={this.props.languages}
                currentLanguage={this.props.currentLanguage}
              />
            </HeaderDropDown>
          </div>
        </div>
      </header>
    )
  }
}

export default Header
