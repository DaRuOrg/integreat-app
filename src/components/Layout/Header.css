.spacer {
  margin-bottom: 100px;
}

.header {
  position: fixed;
  top: 0;
  right: 0;
  left: 0;
  z-index: 1000;

  display: flex;
  height: 100px;
  align-items: center;
  background-color: #FAFAFA;
}

@media screen and (max-width: 600px) {
  .logo {
    display: none;
  }
}

.logo {
  height: 100%;
  padding: 20px;
}

.logo > img {
  height: 100%;
}

.item-active,
.item:hover {
  background-color: #FBDA16;
}

.item {
  display: flex;
  height: 100%;
  max-width: 100px;
  flex-direction: column;
  align-content: center;
  justify-content: center;
  margin: 0 10px;

  font-size: 3rem;
  text-align: center;
}

.item:hover {
  color: inherit;
  text-decoration: inherit;
  cursor: auto;
}

.item-home {
  flex-grow: 1;
  order: 1;
  margin-right: auto;
}

.item-language {
  flex-grow: 1;
  order: 3;
}

.item-location {
  flex-grow: 1;
  order: 2;
}
