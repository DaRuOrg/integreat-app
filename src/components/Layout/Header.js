import React from 'react'
import { NavLink } from 'react-router-dom'
import PropTypes from 'prop-types'
import cx from 'classnames'
import FontAwesome from 'react-fontawesome'
import HeaderDropDown from './HeaderDropDown'
import { isEmpty } from 'lodash/lang'

import style from './Header.css'
import helper from 'components/Helper/Helper.css'
import logo from './assets/integreat-app-logo.png'
import { DEFAULT_NAVIGATION } from 'Navigation'
import LanguageFlyout from 'components/LanguageFlyout'
import { connect } from 'react-redux'

class NavElement extends React.Component {
  static propTypes = {
    to: PropTypes.string.isRequired,
    className: PropTypes.string
  }

  render () {
    return (
      <NavLink exact to={this.props.to} activeClassName={style.itemActive}
               className={cx(this.props.className, helper.removeA)}>
        {this.props.children}
      </NavLink>
    )
  }
}

class Header extends React.Component {
  static propTypes = {
    languageCallback: PropTypes.func,
    currentLanguage: PropTypes.string.isRequired
  }

  render () {
    return (
      <header >
        <div className={style.spacer}>
          <div className={style.header}>
            { /* Logo */}
            <NavElement to={DEFAULT_NAVIGATION.home} className={style.logo}>
              <img src={logo}/>
            </NavElement>
            { /* Location */}
            <NavElement to={DEFAULT_NAVIGATION.location} className={cx(style.item, style.itemLocation)}>
              <FontAwesome name='map-marker'/>
            </NavElement>
            { /* Language */}
            {!isEmpty(this.props.languagePayload.data) &&
            <HeaderDropDown className={style.itemLanguage} fontAwesome="globe">
              <LanguageFlyout
                  languageCallback={this.props.languageCallback}
                  languages={this.props.languagePayload.data}
                  currentLanguage={this.props.currentLanguage}
              />
            </HeaderDropDown>}
          </div>
        </div>
      </header>
    )
  }
}

/**
 * @param state The current app state
 * @returns {{languagePayload: Payload}} The endpoint values from the state mapped to props
 */
function mapeStateToProps (state) {
  return ({
    languagePayload: state.languages
  })
}

export default connect(mapeStateToProps)(Header)
