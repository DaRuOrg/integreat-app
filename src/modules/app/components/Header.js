import React from 'react'
import PropTypes from 'prop-types'
import Headroom from 'react-headroom'
import style from './Header.css'
import logoWide from '../assets/integreat-app-logo.png'
import HeaderMenuBar from './HeaderMenuBar'
import HeaderActionBar from './HeaderActionBar'

class Header extends React.Component {
  static propTypes = {
    menuItems: PropTypes.arrayOf(PropTypes.shape({
      href: PropTypes.string.isRequired, active: PropTypes.bool.isRequired, text: PropTypes.string.isRequired
    })).isRequired,
    actionItems: PropTypes.arrayOf(PropTypes.shape({
      iconSrc: PropTypes.string.isRequired,
      href: PropTypes.string,
      dropDownNode: PropTypes.node
    })).isRequired
  }

  static defaultProps = {
    menuItems: [],
    actionItems: []
  }

  render () {
    return (
      <Headroom>
        <div className={style.header}>
          <div className={style.logoWide}>
            <img src={logoWide} />
          </div>
          <HeaderActionBar items={this.props.actionItems} />
          <HeaderMenuBar className={style.menuBar} items={this.props.menuItems} />
        </div>
      </Headroom>
    )
  }
}

export default Header
