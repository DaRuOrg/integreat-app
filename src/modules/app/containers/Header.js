import React from 'react'
import PropTypes from 'prop-types'
import Headroom from 'react-headroom'
import {translate} from 'react-i18next'
import {connect} from 'react-redux'
import {Link} from 'redux-little-router'
import cx from 'classnames'
import compose from 'lodash/fp/compose'

import Navigation from 'modules/app/Navigation'

import LanguageFlyout from 'modules/app/containers/LanguageFlyout'
import HeaderDropDown from '../components/HeaderDropDown'
import style from './Header.css'
import searchIcon from '../assets/magnifier.svg'
import locationIcon from '../assets/location-icon.svg'
import languageIcon from '../assets/language-icon.svg'
import logoWide from '../assets/integreat-app-logo.png'
import LocationModel from 'modules/endpoint/models/LocationModel'
import withFetcher from 'modules/endpoint/hocs/withFetcher'
import LOCATION_ENDPOINT from 'modules/endpoint/endpoints/location'

class MenuItem extends React.Component {
  static propTypes = {
    href: PropTypes.string.isRequired,
    active: PropTypes.bool
  }

  render () {
    return <Link href={this.props.href} className={cx(style.menuItem, this.props.active ? style.activeMenuItem : '')}>
      {this.props.children}
    </Link>
  }
}

class Header extends React.Component {
  static propTypes = {
    languageCallback: PropTypes.func,
    navigation: PropTypes.instanceOf(Navigation).isRequired,
    location: PropTypes.string,
    locations: PropTypes.arrayOf(PropTypes.instanceOf(LocationModel)),
    route: PropTypes.string.isRequired
  }

  getCurrentLocation () {
    return this.props.locations.find((location) => location.code === this.props.location)
  }

  /**
   * @returns {boolean} True, if there's a location available with events or extras enabled.
   */
  isMenuEnabled () {
    const location = this.getCurrentLocation()
    return !!location && (location.extrasEnabled || location.eventsEnabled)
  }

  isExtrasEnabled () {
    const location = this.getCurrentLocation()
    return location && location.extrasEnabled
  }

  isEventsEnabled () {
    const location = this.getCurrentLocation()
    return location && location.eventsEnabled
  }

  isExtrasSelected () {
    return this.props.route === '/:location/:language/extras'
  }

  isEventsSelected () {
    return this.props.route === '/:location/:language/events*'
  }

  isCategoriesSelected () {
    return this.props.route === '/:location/:language'
  }

  render () {
    const {t} = this.props
    return (
      <Headroom>
        <div className={style.header}>
          <div className={style.logoWide}>
            <img src={logoWide}/>
          </div>
            <div className={style.actionItems}>
            {
              this.props.location &&
              <Link href={this.props.navigation.search} className={style.actionItem}><img src={searchIcon}/></Link>
            }
            <Link href={'/'} className={style.actionItem}><img src={locationIcon}/></Link>
            <HeaderDropDown iconSrc={languageIcon}>
              <LanguageFlyout/>
            </HeaderDropDown>
          </div>
          {
            this.isMenuEnabled() &&
            <div className={style.menuItems}>
              { this.isExtrasEnabled() &&
                <MenuItem href={this.props.navigation.extras} active={this.isExtrasSelected()}>{t('extras')}</MenuItem>
              }
              <MenuItem href={this.props.navigation.home} active={this.isCategoriesSelected()}>{t('categories')}</MenuItem>
              { this.isEventsEnabled() &&
                <MenuItem href={this.props.navigation.events} active={this.isEventsSelected()}>{t('news')}</MenuItem>
              }
            </div>
          }
        </div>
      </Headroom>
    )
  }
}

const mapStateToProps = (state) => ({
  language: state.router.params.language,
  location: state.router.params.location,
  route: state.router.route,
  navigation: new Navigation(state.router.params.location, state.router.params.language)
})

export default compose(
  connect(mapStateToProps),
  withFetcher(LOCATION_ENDPOINT, true, true),
  translate('app')
)(Header)
