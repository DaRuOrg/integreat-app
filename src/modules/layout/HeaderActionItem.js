/**
 * HeaderActionItem is the data class which needs to be supplied to HeaderActionBar.
 */
class HeaderActionItem {
  constructor ({iconSrc, href, dropDownElement}) {
    this._iconSrc = iconSrc
    this._href = href
    this._dropDownElement = dropDownElement
  }

  get iconSrc () {
    return this._iconSrc
  }

  get href () {
    return this._href
  }

  get dropDownElement () {
    return this._dropDownElement
  }
}

export default HeaderActionItem
