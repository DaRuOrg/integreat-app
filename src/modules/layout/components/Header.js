import React from 'react'
import PropTypes from 'prop-types'
import Headroom from 'react-headroom'
import style from './Header.css'
import logoWide from '../assets/integreat-app-logo.png'
import HeaderNavigationBar from './HeaderNavigationBar'
import HeaderActionBar from './HeaderActionBar'
import HeaderActionItem from '../HeaderActionItem'
import HeaderNavigationItem from '../HeaderNavigationItem'

/**
 * The standard header which can supplied to a Layout. Displays a logo left, a HeaderMenuBar in the middle and a
 * HeaderActionBar at the right (RTL: vice versa). On small viewports the HeaderMenuBar is shown underneath the rest
 * of the Header.
 * Uses Headroom to save space when scrolling.
 */
class Header extends React.Component {
  static propTypes = {
    navigationItems: PropTypes.arrayOf(PropTypes.instanceOf(HeaderNavigationItem)).isRequired,
    actionItems: PropTypes.arrayOf(PropTypes.instanceOf(HeaderActionItem)).isRequired
  }

  static defaultProps = {
    navigationItems: [],
    actionItems: []
  }

  render () {
    return (
      <Headroom>
        <header className={style.header}>
          <div className={style.logoWide}>
            <img src={logoWide} />
          </div>
          <HeaderActionBar className={style.actionBar} items={this.props.actionItems} />
          <HeaderNavigationBar className={style.navigationBar} items={this.props.navigationItems} />
        </header>
      </Headroom>
    )
  }
}

export default Header
