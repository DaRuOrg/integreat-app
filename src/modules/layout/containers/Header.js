import React from 'react'
import PropTypes from 'prop-types'
import style from './Header.css'
import logoWide from '../assets/integreat-app-logo.png'
import HeaderNavigationBar from '../components/HeaderNavigationBar'
import HeaderActionBar from '../components/HeaderActionBar'
import HeaderActionItem from '../HeaderActionItem'
import HeaderNavigationItem from '../HeaderNavigationItem'
import Headroom from '../../common/components/Headroom'
import { connect } from 'react-redux'
import { Link } from 'redux-little-router'

const HEADER_HEIGHT_LARGE = 85
const HALF_HEADER_HEIGHT_SMALL = 55

/**
 * The standard header which can supplied to a Layout. Displays a logo left, a HeaderMenuBar in the middle and a
 * HeaderActionBar at the right (RTL: vice versa). On small viewports the HeaderMenuBar is shown underneath the rest
 * of the Header.
 * Uses Headroom to save space when scrolling.
 */
export class Header extends React.Component {
  static propTypes = {
    navigationItems: PropTypes.arrayOf(PropTypes.instanceOf(HeaderNavigationItem)).isRequired,
    actionItems: PropTypes.arrayOf(PropTypes.instanceOf(HeaderActionItem)).isRequired,
    smallViewport: PropTypes.bool.isRequired,
    logoHref: PropTypes.string.isRequired
  }

  static defaultProps = {
    navigationItems: [],
    actionItems: []
  }

  render () {
    const scrollHeight = this.props.smallViewport ? HALF_HEADER_HEIGHT_SMALL : HEADER_HEIGHT_LARGE
    return (
      <Headroom scrollHeight={scrollHeight}>
        <header className={style.header}>
          <div className={style.logoWide}>
            <Link href={this.props.logoHref}>
              <img src={logoWide} />
            </Link>
          </div>
          <HeaderActionBar className={style.actionBar} items={this.props.actionItems} />
          <HeaderNavigationBar className={style.navigationBar} items={this.props.navigationItems} />
        </header>
      </Headroom>
    )
  }
}

const mapStateToProps = state => ({smallViewport: state.viewport.is.small})
export default connect(mapStateToProps)(Header)
