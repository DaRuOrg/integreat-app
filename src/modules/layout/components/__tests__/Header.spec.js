import React from 'react'
import HeaderActionItem from '../../HeaderActionItem'
import HeaderNavigationItem from '../../HeaderNavigationItem'
import Header from '../Header'
import { shallow } from 'enzyme'

describe('Header', () => {
  test('should match snapshot', () => {
    const component = shallow(
      <Header actionItems={[new HeaderActionItem({href: '/action1'})]}
              navigationItems={[new HeaderNavigationItem({href: '/nav1'})]} />
    )
    expect(component).toMatchSnapshot()
  })
})
