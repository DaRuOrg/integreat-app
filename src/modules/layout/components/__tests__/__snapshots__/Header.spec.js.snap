// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match snapshot 1`] = `
<Headroom
  calcHeightOnResize={true}
  disable={false}
  disableInlineStyles={false}
  downTolerance={0}
  onPin={[Function]}
  onUnfix={[Function]}
  onUnpin={[Function]}
  parent={[Function]}
  pinStart={0}
  upTolerance={5}
  wrapperStyle={Object {}}
>
  <header
    className="header"
  >
    <div
      className="logoWide"
    >
      <img
        src="test-file-stub"
      />
    </div>
    <HeaderActionBar
      className="actionBar"
      items={
        Array [
          HeaderActionItem {
            "_dropDownNode": undefined,
            "_href": "/action1",
            "_iconSrc": undefined,
          },
        ]
      }
    />
    <HeaderMenuBar
      className="navigationBar"
      items={
        Array [
          HeaderNavigationItem {
            "_active": undefined,
            "_href": "/nav1",
            "_text": undefined,
          },
        ]
      }
    />
  </header>
</Headroom>
`;
